kion/internal/kionclient/client.go

```diff
@@ -132,66 +141,21 @@ func (client *Client) POST(urlPath string, sendData interface{}) (*Creation, err
 		return nil, err
 	}
 
-	data := Creation{}
-	err = json.Unmarshal(body, &data)
-	if err != nil {
+	var data Creation
+	if err := json.Unmarshal(body, &data); err != nil {
 		return nil, fmt.Errorf("could not unmarshal response body: %v", string(body))
 	}
```

Comment:
It's odd that we are populating a creation struct for PUTs and PATCHes.  I understand it will still "work" in these scenarios, since all PUTs and PATCHes return a JSON string that will partly marshal into the Creation object (albeit without the record id), and then the PUT and PATCH methods ignore the response.  But this seems like a case of [DRYing prematurely](https://testing.googleblog.com/2024/05/dont-dry-your-code-prematurely.html).

What is the ultimate value in combining the logic for POST, PUT and PATCH?  Is it just DRY?  Because if so, I'd argue that POST had different behavior that is now basked into PUT and PATCH unnecessarily.

Reply:
The goal was to be DRY and streamline the request creation process, and I didn't think too much of a consequence of the methods just ignoring the response. What you say makes sense.  TIL that 'DRYing prematurely' is a thing, but of course, it is...

Reply:
It stemmed from a lint finding where the HTTP request was inefficient or something, and so I figured, why not just combine all 3?

Reply:
[@https://github.com/kionsoftware/terraform-provider-kion/pull/69/commits/34c4a94941a823d0364b79db57930365ad749e77](https://github.com/kionsoftware/terraform-provider-kion/pull/69/commits/34c4a94941a823d0364b79db57930365ad749e77)